src/Bitmap/BitmapFile.h

```diff
@@ -35,6 +35,7 @@ class BitmapFile
 	static void WriteIndexed(std::string filename, uint16_t bitCount, int32_t width, int32_t height, std::vector<Color> palette, const std::vector<uint8_t>& indexedPixels);
 	static void WriteIndexed(Stream::BidirectionalWriter& seekableWriter, uint16_t bitCount, int32_t width, int32_t height, std::vector<Color> palette, const std::vector<uint8_t>& indexedPixels);
 	static void WriteIndexed(std::string filename, const BitmapFile& bitmapFile);
+	void WriteIndexed(Stream::BidirectionalWriter& writer) const;
```

Comment:
Is bidirectional really needed here? For reading, the bidirectional helps with peeking to determine the bitmap type. I don't think it would be required for writing though. Should be able to write out the whole bitmap in one go.

Reply:
The underlying static function behind this call takes a bidirectional writer. Although you are right it doesn't need to seek. Stemming from sloppy work by me in the past.

Sounds like we are on the same page to move from static to member function calls for WriteIndexed. Will plan to switch to WriteIndexed taking a simple Writer interface at that point as well.